### PR TITLE
Pygame input

### PR DESCRIPTION
**Pull request by SevakAbr**

### CODE REVIEW

**Review by fisheeve — APPROVED** [review]

File: game/player.py [hunk 1]
```diff
@@ -13,16 +19,32 @@ class KeyboardPlayer(Player):
     def __init__(self, stats: Stats):
         super().__init__(stats)
         self.key_pressed = None
+        pygame.init()
+        listener = threading.Thread(target=self.listener)
```
Comment (zidder) [inline game/player.py]: `daemon=True`

File: game/player.py [hunk 2]
```diff
@@ -13,16 +19,32 @@ class KeyboardPlayer(Player):
...
     def decide(self):
-        if self.key_pressed is not None:
-            return self.key_pressed
+        keys = self.key_pressed
+        self.key_pressed = []
+        if keys is not None:
+            return keys
         else:
-            return Action.NOTHING
+            return [Action.NOTHING]
```
Comment (zidder) [inline game/player.py]: You can return `[]` as well :)

File: game/player.py [hunk 3]
```diff
@@ -13,16 +19,32 @@ class KeyboardPlayer(Player):
...
 
     def listener(self):
         while True:
-            self.key_pressed = self.keyboard_listener.getkey()
+            keys = pygame.key.get_pressed()
+            if keys[pygame.K_LEFT]:
```
Comment (zidder) [inline game/player.py:30]: Don't you need to initialize `key_pressed` with two `NOTHING` items before doing this?

File: game/player.py [hunk 4]
```diff
@@ -13,16 +19,32 @@ class KeyboardPlayer(Player):
...
+                self.key_pressed[0] = Action.MOVE_LEFT
+            elif keys[pygame.K_RIGHT]:
+                self.key_pressed[0] = Action.MOVE_RIGHT
+            elif keys[pygame.K_DOWN]:
+                self.key_pressed[0] = Action.MOVE_DOWN
+            elif keys[pygame.K_UP]:
+                self.key_pressed[0] = Action.MOVE_UP
+            elif keys[pygame.K_SPACE]:
+                self.key_pressed[1] = Action.SHOOT
+            # pygame.event.pump()
```
Comment (zidder) [inline game/player.py:40]: What does this do?

**Review by zidder — CHANGES_REQUESTED** [review]

File: game/player.py [hunk 5]
```diff
@@ -13,16 +15,29 @@ class KeyboardPlayer(Player):
     def __init__(self, stats: Stats):
         super().__init__(stats)
         self.key_pressed = None
```
Comment (zidder) [inline game/player.py]: Here as well `Nothing, Nothing`

**Review by zidder — CHANGES_REQUESTED** [review]

**Review by zidder — APPROVED** [review]

**Pull request merged**